src/Symfony/Component/Translation/Command/XliffLintCommand.php

```diff
@@ -115,14 +117,16 @@ private function validate($content, $file = null)
         $document->loadXML($content);
 
         if (null !== $targetLanguage = $this->getTargetLanguageFromFile($document)) {
-            $expectedFileExtension = sprintf('%s.xlf', str_replace('-', '_', $targetLanguage));
-            $realFileExtension = explode('.', basename($file), 2)[1] ?? '';
+            $normalizedLocale = preg_quote(str_replace('-', '_', $targetLanguage));
```

Comment:
you need to pass the regex delimiter as second argument to `preg_quote` as it needs to be escaped too.